### PR TITLE
Clarify Mac/Linux NuGet.Config file locations in configuring-nuget-behavior.md

### PR DESCRIPTION
**Pull request by arykov**
For discussion of this, please see https://github.com/dotnet/sdk/issues/12674. Note, I did not validate on Mac

### CODE REVIEW

**Comment by arykov** [comment]
@dotnet-policy-service agree

**Review by JonDouglas — APPROVED** [review]
LGTM, @heng-liu can you confirm/review?

**Review by heng-liu — COMMENTED** [review]
The code shows `MachineWideConfigDirectory` is `MachineWideSettingsBaseDirectory` + `Config`
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L45-L48

And on Linux/Mac, the `MachineWideSettingsBaseDirectory` is `SpecialFolder.CommonApplicationData` + `NuGet`
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L38-L43C26

**In core CLR**(e.g. using dotnet.exe),
On MacOS, `SpecialFolder.CommonApplicationData` is set to `/Library/Application Support`
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L175-L177
On Linux,  if env var `NUGET_COMMON_APPLICATION_DATA` is not null or empty, `SpecialFolder.CommonApplicationData` is set to the path `NUGET_COMMON_APPLICATION_DATA` points to, otherwise, it will is set to `/etc/opt` :
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L181-L188

**In full framework code path**(e.g. using Mono on Linux/MacOS),
 On Linux/MacOS, the `SpecialFolder.CommonApplicationData` is set to [Environment.SpecialFolder.CommonApplicationData](https://learn.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=netframework-4.7.2)
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L254-L256

**Comment by heng-liu** [comment]
I think all the complication is because in the past, there was no `Environment.SpecialFolder` available in .NET Core when NuGet implements this :( https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L396-L398

**Comment by arykov** [comment]
@heng-liu, what are the implications of your comment for purposes of my update?  Do you want me to restate it somehow?

**Comment by heng-liu** [comment]
Hi @arykov , I'm sorry for losing track on this for so long!
Please update the document based on the comments. Thanks!

File: docs/consume-packages/configuring-nuget-behavior.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ NuGet's behavior is driven by the accumulated settings in one or more `NuGet.Con
...
 | User | **Windows:** `%appdata%\NuGet\NuGet.Config`<br/>**Mac/Linux:** `~/.config/NuGet/NuGet.Config` or `~/.nuget/NuGet/NuGet.Config` (varies by tooling) <br/>Additional configs are supported on all platforms. These configs cannot be edited by the tooling. </br> **Windows:** `%appdata%\NuGet\config\*.Config` <br/>**Mac/Linux:** `~/.config/NuGet/config/*.config` or `~/.nuget/config/*.config` | Settings apply to all operations, but are overridden by any solution-level settings. |
-| Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. If `$XDG_DATA_HOME` is null or empty, `~/.local/share` or `/usr/local/share` will be used (varies by OS distribution)  | Settings apply to all operations on the computer, but are overridden by any user- or solution-level settings. |
+| Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `$XDG_DATA_HOME`. /etc/opt/NuGet/Config by default. If $NUGET_COMMON_APPLICATION_DATA is neither null nor empty, then $NUGET_COMMON_APPLICATION_DATA/NuGet/Config instead  | Settings apply to all operations on the computer, but are overridden by any user- or solution-level settings. |
```
Comment (lucastheisen) [inline docs/consume-packages/configuring-nuget-behavior.md:20]: Isn't any mention of `XDG_DATA_HOME` incorrect?  The way i read the commentary and the source its not mentioned at all in the context of locating the `NuGet.Config` file.  So shouldn't this be:

```suggestion
| Computer | **Windows:** `%ProgramFiles(x86)%\NuGet\Config`<br/>**Mac/Linux:** `/etc/opt/NuGet/Config` (Linux) or `/Library/Application Support` (Mac) by default. If `$NUGET_COMMON_APPLICATION_DATA` is neither null nor empty, then `$NUGET_COMMON_APPLICATION_DATA/NuGet/Config` instead  | Settings apply to all operations on the computer, but are overridden by any user- or solution-level settings. |
```

Comment (arykov) [reply docs/consume-packages/configuring-nuget-behavior.md:20]: @heng-liu, please, rephrase as you deem fit. I am not 100% sure which aspects of proposed changes you found inaccurate or incomplete.

**Review by donnie-msft — APPROVED** [review]

**Pull request merged**